### DISCUSSION TITLE
Beginer question: BMS is talking, how to see data in HA

### DISCUSSION THREAD

**Discussion by Battlewind** [syssi/esphome-jbd-bms#65] [General]
I have the ESP32 board talking and communicating with the BMS. However, I don't understand where I will get the data from the HA system. Is it an addon? Will it show up in energy? I am using the API instead of MQTT. What am I missing as I can't find it in the beginners document.

**Comment by syssi** [comment]
There is a Home Assistant integration called `ESPHome`. Please go to:

Settings -> Devices & services -> "+ Add integration" -> Search for "ESPHome"

If you ESPHome was auto-discovered select the item from the "Discovered" list. If the list is empty you can add the node manually using "Setup another instance of ESPHome". As `Host` it's recommended to use the Hostname of the node + the `.local` suffix. 

As soon the node is added to your ESPHome setup you should find a new devices showing all the exposed measurements.

To use the energy dashboard you have to extend the YAML because an additional sensor (`charging energy today`, `discharging energy today`) is required. Do you use a UART-TTL connection or BLE? I will provide a small example as soon you see the measurements at your HA instance.

**Reply by Battlewind** [reply]
I have the ESPHome integration installed. I do the initial basic install of the ESP32 board then I edit the yaml file and put in all the data from the repository for my BMS, then change the MAC address, then remove the MQTT and uncomment the API. Is that the correct order?

I am using BLE.

**Reply by syssi** [reply]
Most people mix up the `ESPHome add-on` vs. `ESPHome integration`. I guess you used the Add-On to flash the YAML configuration to the ESP. You should be able to take a look at the logbook of the ESPHome node using the Add-on.

The Home Assistant integration called `ESPHome` is the server of the native API. Did you add this integration to integrate the new ESPHome device?

**Reply by Battlewind** [reply]
Okay. So now that I posted here, it seems to be starting to work. I tried to change over to the multi BMS code and it didn't. If I wanted to stick with the resolution on the single BMS, but poll both BMS at the same time, would I use the yaml file twice?

**Reply by syssi** [reply]
Please compare the multi BMS YAML to the single one. You will identify different pieces / components which needs to be instantiated multi times f.e. a second `ble_client` connection and a second loaded `jbd_bms_ble` component which listens on the second BLE connection. If this doesn't work properly for you please use two ESPs and write the single BMS YAML to both. Just the `mac_address` and the node / hostname must be different.

**Reply by Battlewind** [reply]
I will probably stick to a second ESP board. I am a little leary of one board taking out two BMS readbacks at one time. Thanks again for you help. You have been awesome!
